src/AggregateFunctions/AggregateFunctionGroupArray.cpp

```diff
@@ -457,7 +457,8 @@ struct GroupArrayNodeGeneral : public GroupArrayNodeBase<GroupArrayNodeGeneral>
     static Node * allocate(const IColumn & column, size_t row_num, Arena * arena)
     {
         const char * begin = arena->alignedAlloc(sizeof(Node), alignof(Node));
-        auto value = column.serializeAggregationStateValueIntoArena(row_num, *arena, begin);
+        IColumn::SerializationSettings settings{.serialize_string_with_zero_byte = true};
```

Comment:
Let's write a comment that we enable `serialize_string_with_zero_byte` in such places because of compatibility. Or maybe create a separate function like `createSerializationSettingsForAggregationState` with such comment and use it in all places. WDYT?